src/views/Posts/Posts.tsx

```diff
@@ -1,21 +1,33 @@
 import { Box } from '@mui/system';
-import React, { useEffect } from 'react';
-import { useLocation, useParams } from 'react-router';
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router';
+import { getUser } from '../../api/users';
 import Header from '../../components/Header/Header';
 import PostCard from '../../components/PostCard/PostCard';
 import { getPostsList } from '../../redux/actions/posts';
 import { useAppDispatch, useAppSelector } from '../../redux/hooks';
 
 const Posts = () => {
   const params = useParams();
-  const { state } = useLocation();
+  const [username, setUsername] = useState<string>('');
+  const [userEmail, setEmail] = useState<string>('');
+  const navigate = useNavigate();
   const { info, isProcessing } = useAppSelector((state) => state.postReducer);
   const dispatch = useAppDispatch();
 
+  const getUserDetails = async () => {
```

Comment:
Add type

Reply:
I've added a type to getUserDetails and is it the correct way?